server/src/test/java/org/elasticsearch/index/mapper/BooleanFieldMapperTests.java

```diff
@@ -202,14 +202,19 @@ public void testScriptAndPrecludedParameters() {
         assertThat(e.getMessage(), equalTo("Failed to parse mapping: Field [null_value] cannot be set in conjunction with field [script]"));
     }
 
+    @Override
+    protected List<ExampleMalformedValue> exampleMalformedValues() {
+        return List.of(exampleMalformedValue("a").errorMatches("Failed to parse value [a] as only [true] or [false] are allowed."));
+    }
+
     @Override
     protected boolean supportsIgnoreMalformed() {
-        return false;
+        return true;
     }
 
     @Override
     protected SyntheticSourceSupport syntheticSourceSupport(boolean ignoreMalformed) {
-        assertFalse("boolean doesn't support ignore_malformed", ignoreMalformed);
+        assumeFalse("boolean doesn't support ignore_malformed", ignoreMalformed);
```

Comment:
I might update the error message to "boolean doesn't support ignore_malformed with synthetic source". But otherwise, I'm good with the `assumeFalse` here.